### ISSUE TITLE
Issues at High Zoom

### ISSUE THREAD

**Issue by bclehmann** [ScottPlot/ScottPlot#1873]
## Bug Report

**Issue:** These are all instances of #1849, but visible on other plottables

**Reproducing:** (What do others have to do to reproduce the bug?)

I can replicate these all by zooming very far into filled regions for these plot-types:
- Fill
- Polygon
- Coxcomb
- Radar
- Candlestick
- OHLC
- Heatmap
- ScaleBar
- Scatter
- Signal
- Arrow
- VectorField

It appears it's also possible for lines (not fills), as you can see in this screenshot from an OHLC
<img width="332" alt="image" src="https://user-images.githubusercontent.com/8635304/172091520-c915e580-0743-497b-b7c4-ed83d6e4cbea.png">
<img width="329" alt="image" src="https://user-images.githubusercontent.com/8635304/172091499-7824e08d-9ceb-4319-a38f-8143ae962229.png">

Additionally:
- RadialGauge seems to crash on high zoom

### System Details
* **ScottPlot Version:** 4.1.46
* **Operating System:** Windows 11
* **Application Type:** WPF
* **.NET Version:** .NET Framework 4.8

**Comment by bclehmann** [comment]
In my opinion, since this occurs when the user zooms in so far that the nearest datapoints are very far off-screen, I don't think this is likely to be worth fixing just on its own. If the datapoints are far off-screen the expected behaviour is a useless graph, which you could argue is what the user gets.

Of course, if we're already touching an affected plottable it then would make sense to also clip what we're drawing.

As for the crash in RadialGauge, here is the callstack
```
System.Runtime.InteropServices.ExternalException
  HResult=0x80004005
  Message=A generic error occurred in GDI+.
  Source=System.Drawing
  StackTrace:
   at System.Drawing.Graphics.CheckErrorStatus(Int32 status)
   at System.Drawing.Graphics.DrawString(String s, Font font, Brush brush, RectangleF layoutRectangle, StringFormat format)
   at System.Drawing.Graphics.DrawString(String s, Font font, Brush brush, Single x, Single y, StringFormat format)
   at ScottPlot.Plottable.RadialGauge.RenderGaugeLabels(Graphics gfx, PlotDimensions dims, PointF center, Single radius) in /_/src/ScottPlot4/ScottPlot/Plottable/RadialGauge.cs:line 211
   at ScottPlot.Plottable.RadialGauge.Render(Graphics gfx, PlotDimensions dims, PointF centerPixel, Single radius) in /_/src/ScottPlot4/ScottPlot/Plottable/RadialGauge.cs:line 125
   at ScottPlot.Plottable.RadialGaugePlot.Render(PlotDimensions dims, Bitmap bmp, Boolean lowQuality) in /_/src/ScottPlot4/ScottPlot/Plottable/RadialGaugePlot.cs:line 311
   at ScottPlot.Plot.RenderPlottables(Bitmap bmp, Boolean lowQuality, Double scaleFactor) in /_/src/ScottPlot4/ScottPlot/Plot/Plot.Render.cs:line 81
   at ScottPlot.Plot.Render(Bitmap bmp, Boolean lowQuality, Double scale) in /_/src/ScottPlot4/ScottPlot/Plot/Plot.Render.cs:line 43
   at ScottPlot.Control.ControlBackEnd.Render(Boolean lowQuality, Boolean skipIfCurrentlyRendering) in /_/src/ScottPlot4/ScottPlot/Control/Backend.cs:line 337
   at ScottPlot.Control.ControlBackEnd.ProcessEvent(IUIEvent uiEvent) in /_/src/ScottPlot4/ScottPlot/Control/Backend.cs:line 600
   at ScottPlot.Control.ControlBackEnd.MouseWheel(InputState input) in /_/src/ScottPlot4/ScottPlot/Control/Backend.cs:line 713
   at ScottPlot.WpfPlot.OnMouseWheel(Object sender, MouseWheelEventArgs e) in C:\Users\benny\source\repos\ScottPlot\src\ScottPlot4\ScottPlot.WPF\WpfPlot.xaml.cs:line 210
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Window.ShowHelper(Object booleanBox)
   at System.Windows.Window.ShowDialog()

  This exception was originally thrown at this call stack:
    [External Code]
    ScottPlot.Plottable.RadialGauge.RenderGaugeLabels(System.Drawing.Graphics, ScottPlot.PlotDimensions, System.Drawing.PointF, float) in RadialGauge.cs
    ScottPlot.Plottable.RadialGauge.Render(System.Drawing.Graphics, ScottPlot.PlotDimensions, System.Drawing.PointF, float) in RadialGauge.cs
    ScottPlot.Plottable.RadialGaugePlot.Render(ScottPlot.PlotDimensions, System.Drawing.Bitmap, bool) in RadialGaugePlot.cs
    ScottPlot.Plot.RenderPlottables(System.Drawing.Bitmap, bool, double) in Plot.Render.cs
    ScottPlot.Plot.Render(System.Drawing.Bitmap, bool, double) in Plot.Render.cs
    ScottPlot.Control.ControlBackEnd.Render(bool, bool) in Backend.cs
    ScottPlot.Control.ControlBackEnd.ProcessEvent(ScottPlot.Control.EventProcess.IUIEvent) in Backend.cs
    ScottPlot.Control.ControlBackEnd.MouseWheel(ScottPlot.Control.InputState) in Backend.cs
    ScottPlot.WpfPlot.OnMouseWheel(object, System.Windows.Input.MouseWheelEventArgs) in WpfPlot.xaml.cs
    ...
    [Call Stack Truncated]
```

Since the error comes immediately applying a translation (with high values due to the zoom), I suspect it's again because we're drawing something far off-screen.

**Comment by bclehmann** [comment]
Also sorry for opening this right after 4.1.46 released, I only saw #1849 when I read the changelog, and I figured it was probably something that I could replicate on other plottables.

**Comment by swharden** [comment]
> Also sorry for opening this right after 4.1.46 released

No worries!

> In my opinion, since this occurs when the user zooms in so far that the nearest datapoints are very far off-screen, I don't think this is likely to be worth fixing just on its own. If the datapoints are far off-screen the expected behaviour is a useless graph, which you could argue is what the user gets.

I agree here. If the user zooms in that much, they're being weird lol. At its core this is a System.Drawing bug, and I'm looking forward to leaving this one behind us after moving to the next thing 🤞 #1036
